src/main/java/org/kairosdb/core/http/rest/MetricsResource.java

```diff
@@ -543,6 +548,13 @@ public Response get(@Context HttpHeaders httpHeaders, String json) throws Except
 			span.log(e.getMessage());
 			return setHeaders(Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(new ErrorResponse(e.getMessage()))).build();
 		}
+		catch (TimeoutException e)
+		{
+			logger.error("Request timed out at " + datastore.getM_readTimeout() + " milli seconds", e);
```

Comment:
Since this is the generic TimeoutException, it could be possible that this could be thrown in other unexpected ways.  We may want to specialize our exception in this case to be sure we catch only exactly what we want here.